src/components/AllWorkOrders.js

```diff
@@ -0,0 +1,193 @@
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faTruck, faMapMarkerAlt } from '@fortawesome/free-solid-svg-icons'
+
+import api from '../queries/api'
+import { workOrderFields } from '../queries/fields'
+import { signalsWorkOrderStatuses } from '../constants/statuses'
+
+const fields = workOrderFields.baseFields
+const statuses = signalsWorkOrderStatuses
+
+class AllWorkOrders extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            allWorkOrdersData: [],
+            location: '',
+            currentPage: 1,
+            lastPage: 1,
+        }
+    }
+
+    componentDidMount() {
+        api.allWorkOrders()
+            .getAll()
+            .then(res => {
+                this.setState({
+                    allWorkOrdersData: res.data.records,
+                    lastPage: res.data.total_pages,
+                })
+                console.log(res.data)
+            })
+    }
+
+    handleChange = event => {
+        this.setState({
+            location: event.target.value,
+        })
+    }
+
+    handleSubmit = event => {
+        event.preventDefault()
+        api.allWorkOrders()
+            .searchAll(this.state.location, 1)
+            .then(res => {
+                this.setState({
+                    allWorkOrdersData: res.data.records,
+                    lastPage: res.data.total_pages,
+                    currentPage: 1,
+                })
+                console.log(res.data)
+            })
+    }
+
+    updatePage = pageNumber => {
+        api.allWorkOrders()
+            .searchAll(this.state.location, pageNumber)
+            .then(res => {
+                this.setState({
+                    allWorkOrdersData: res.data.records,
+                    lastPage: res.data.total_pages,
+                    currentPage: pageNumber,
+                })
+                console.log(res.data)
+            })
+        window.scrollTo(0, 0)
```

Comment:
Awesome job with the pagination! And catching details like the scroll back to top.

I wonder what it would take to animate the scrollTo top. And also wondering if we should return fewer items on each page. Probably not, just thinking as a type...

Reply:
Thanks! It was pretty fun since I've never added pagination to an app. The API response helped me to see what to do with the pages (thanks Knack!). I agree about the animation so I'll look into it so it isn't so abrupt. I added updating the state.loading boolean in updatePage() as well as setting empty state for allWorkOrdersData so the user sees that data is loading when switching pages.

I thought 100 records seemed like a lot too. I almost tried 50 since there is pagination, but I also don't want the user to have to page over too much. It can easily be changed in the API call so we can think about it.